### PR TITLE
remove SetKeychainSecurityGroup

### PR DESCRIPTION
**Pull request by jennyf19**
- ITokenCacheAccessor
- TokenCacheAccessor
- IPublicClientApp
- iOSTokenCacheAccessor
- PublicClientApp
- TelemetryTokenCacheAccessor
fix spelling

### CODE REVIEW

File: src/Microsoft.Identity.Client/IPublicClientApplication.cs [hunk 1]
```diff
@@ -150,15 +150,6 @@ public partial interface IPublicClientApplication : IClientApplicationBase
...
         string iOSKeychainSecurityGroup { get; set; }
-
-        /// <summary>
-        /// Xamarin iOS specific property enabling the application to share the token cache with other applications sharing the same keychain security group.
-        /// If you use this property, you MUST add the capability to your Application Entitlement.
-        /// When using this property, the value must contain the TeamId prefix, which is why this will be obsolete in future releases.
-        /// </summary>
-        /// <remarks>This API will be removed in MSAL v3.x. See https://aka.ms/msal-net-ios-keychain-security-group for details</remarks>
-        [Obsolete("Use iOSKeychainSecurityGroup instead (See https://aka.ms/msal-net-ios-keychain-security-group)", false)]
-        string KeychainSecurityGroup { get; set; }
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/IPublicClientApplication.cs]: was this added as migration aid instead?

File: src/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 2]
```diff
@@ -121,24 +121,6 @@ public string iOSKeychainSecurityGroup
                 UserTokenCacheInternal.SetIosKeychainSecurityGroup(value);
             }
         }
-
-        /// <inheritdoc />
-        [Obsolete("Use iOSKeychainSecurityGroup instead (See https://aka.ms/msal-net-ios-keychain-security-group)", false)]
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/PublicClientApplication.cs]: >[Obsolete("Use iOS [](start = 8, length = 18)

pls. see comment on the interface

File: src/Microsoft.Identity.Client/IPublicClientApplication.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Client/IPublicClientApplication.cs]: #no....but will include

**Comment by henrik-me** [comment]
Generally surprised that there are no utests to remove? Is the new method appropriately covered with utests?

File: src/Microsoft.Identity.Client/IPublicClientApplication.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Client/IPublicClientApplication.cs]: have it here: [https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/817](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/817)

---
In reply to: [249953245](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/814#discussion_r249953245) [](ancestors = 249953245)

**Comment by jennyf19** [comment]
this is because it is iOS specific. i could look into adding in a UI automation test for this, but not sure how app center will react. I have it in the dev xamarin ios app, so can be tested manually for now.

---
In reply to: [456557203](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/814#issuecomment-456557203) [](ancestors = 456557203)

**Pull request merged**